homeassistant/components/keymitt_ble/config_flow.py

```diff
@@ -0,0 +1,164 @@
+        # This is for backwards compatibility.
+        return await self.async_step_init(user_input)
+
+    async def async_step_init(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Check if paired."""
+        errors: dict[str, str] = {}
+
+        if discovery := self._discovered_adv:
+            self._discovered_advs[discovery.address] = discovery
+            self._name = name_from_discovery(discovery)
+        else:
+            self._name = DEFAULT_NAME
+            current_addresses = self._async_current_ids()
+            for discovery_info in async_discovered_service_info(self.hass):
+                address = discovery_info.address
+                if (
+                    address in current_addresses
+                    or address in self._discovered_advs
+                ):
+                    continue
+                parsed = parse_advertisement_data(
+                    discovery_info.device, discovery_info.advertisement
+                )
+                if parsed:
+                    self._discovered_advs[address] = parsed
+
+        if not self._discovered_advs:
+            return self.async_abort(reason="no_unconfigured_devices")
+
+        data_schema = vol.Schema(
+            {
+                vol.Required(CONF_BDADDR): vol.In(
+                    {
+                        address: f"{parsed.data['local_name']} ({address})"
+                        for address, parsed in self._discovered_advs.items()
+                    }
+                )
+            }
+        )
+
+        if user_input is not None:
+            self._bdaddr = user_input[CONF_BDADDR]
+            await self.async_set_unique_id(
+                self._bdaddr, raise_on_progress=False
+            )
+            self._abort_if_unique_id_configured()
+            self._ble_device = bluetooth.async_ble_device_from_address(
+                self.hass, self._bdaddr
+            )
+            if not self._ble_device:
+                raise IntegrationError(
+                    f"Could not find MicroBot with address {self._bdaddr}"
+                )
+            return await self.async_step_link()
+
+        return self.async_show_form(
+            step_id="init", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_link(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Given a configured host, will ask the user to press the button to pair."""
+        errors: dict[str, str] = {}
+        token = randomid(32)
+        self._client = MicroBotApiClient(
+            device=self._ble_device,
+            token=token,
+        )
+        assert self._client is not None
+        if user_input is None:
+            await self._client.connect(init=True)
+            return self.async_show_form(step_id="link")
+
+        if not self._client.is_connected():
+            errors["base"] = "linking"
```

Comment:
Usually we use the default `cannot_connect` error for this case.  You'll need to update strings.json as well to use the common key as well

Reply:
I'm using is_connected at this point to establish if the linking of the token is successful. If it can't connect then the token linking has failed